singular-kit/src/main/java/com/mparticle/kits/SingularKit.java

```diff
@@ -0,0 +1,172 @@
+        config = new SingularConfig(singularKey, singularSecret);
+        config.withDDLTimeoutInSec(DDL_HANDLER_TIMEOUT_SEC);
+        config.withDDLHandler(this);
+        Singular.init(context, config);
+        return null;
+    }
+
+    @Override
+    public List<ReportingMessage> setOptOut(boolean b) {
+        return null;
+    }
+
+    @Override
+    public String getName() {
+        return "Singular";
+    }
+
+    @Override
+    public List<ReportingMessage> onActivityCreated(Activity activity, Bundle bundle) {
+        return null;
+    }
+
+    @Override
+    public List<ReportingMessage> onActivityStarted(Activity activity) {
+        return null;
+    }
+
+    @Override
+    public List<ReportingMessage> onActivityResumed(Activity activity) {
+        Singular.onActivityResumed();
+        return null;
+    }
+
+    @Override
+    public List<ReportingMessage> onActivityPaused(Activity activity) {
+        Singular.onActivityPaused();
+        return null;
+    }
+
+    @Override
+    public List<ReportingMessage> onActivityStopped(Activity activity) {
+        return null;
+    }
+
+    @Override
+    public List<ReportingMessage> onActivitySaveInstanceState(Activity activity, Bundle bundle) {
+        return null;
+    }
+
+    @Override
+    public List<ReportingMessage> onActivityDestroyed(Activity activity) {
+        return null;
+    }
+
+    @Override
+    public List<ReportingMessage> leaveBreadcrumb(String s) {
+        return null;
+    }
+
+    @Override
+    public List<ReportingMessage> logError(String s, Map<String, String> map) {
+        return null;
+    }
+
+    @Override
+    public List<ReportingMessage> logException(Exception e, Map<String, String> map, String s) {
+        return null;
+    }
+
+    @Override
+    public List<ReportingMessage> logEvent(MPEvent mpEvent) {
+        String eventName = mpEvent.getEventName();
+        Map eventInfo = mpEvent.getInfo();
+        if (eventInfo != null) {
+            JSONObject params = new JSONObject(eventInfo);
+            Singular.event(eventName, params.toString());
+        } else {
+            Singular.event(eventName);
+        }
+        return null;
+    }
+
+    @Override
+    public List<ReportingMessage> logScreen(String s, Map<String, String> map) {
+        return null;
+    }
+
+    @Override
+    public void handleLink(String link) {
+        DeepLinkListener deepLinkListener = MParticle.getInstance().getDeepLinkListener();
+        if (deepLinkListener != null) {
+            DeepLinkResult deepLinkResult = new DeepLinkResult();
+            deepLinkResult.setServiceProviderId(MParticle.ServiceProviders.SINGULAR);
+            deepLinkResult.setLink(link);
+            deepLinkListener.onResult(deepLinkResult);
+        }
+    }
+
+    @Override
+    public boolean willHandlePushMessage(Intent intent) {
+        return false;
+    }
+
+    @Override
+    public void onPushMessageReceived(Context context, Intent intent) {
+
```

Comment:
Similarly to above, if you support displaying of push notification, you should initialize your push receiver here and pass the intent into it.